core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/ImpersonationWarning.scala

```diff
@@ -0,0 +1,183 @@
+      ldapWarning(rule.name, blockName, rule.settings.ldap.id, rule.impersonation)(requestId)
+    }
+    implicit val ldapAuthRule: ImpersonationWarningSupport[LdapAuthRule] = WithPossibleWarnings[LdapAuthRule] { (rule, blockName, requestId) =>
+      ldapWarning(rule.name, blockName, rule.authorization.settings.ldap.id, rule.authorization.impersonation)(requestId)
+    }
+    implicit val rorKbnAuthRule: ImpersonationWarningSupport[RorKbnAuthRule] = WithPossibleWarnings[RorKbnAuthRule] { (rule, blockName, _) =>
+      Some(impersonationNotSupportedWarning(rule, blockName))
+    }
+
+    private def fromHashedCredentials[R <: BasicAuthenticationRule[HashedCredentials]](rule: R, blockName: Block.Name, requestId: RequestId): Option[ImpersonationWarning] = {
+      rule.settings.credentials match {
+        case _: HashedCredentials.HashedUserAndPassword =>
+          Some(ImpersonationWarning(
+            block = blockName,
+            ruleName = rule.name,
+            message = "The rule contains fully hashed username and password. It doesn't support impersonation in this configuration",
+            hint = s"You can use second version of the rule and use not hashed username. Like that: `${rule.name.value}: USER_NAME:hash(PASSWORD)"
+          ))
+        case _: HashedCredentials.HashedOnlyPassword =>
+          None
+      }
+    }
+
+    private def impersonationNotSupportedWarning[R <: Rule](rule: R, blockName: Block.Name) =
+      ImpersonationWarning(
+        block = blockName,
+        ruleName = rule.name,
+        message = "Impersonation is not supported by this rule by default",
+        hint = ""
+      )
+
+    private def ldapWarning(ruleName: Rule.Name,
+                            blockName: Block.Name,
+                            serviceId: LdapService#Id,
+                            impersonation: Impersonation)
+                           (implicit requestId: RequestId): Option[ImpersonationWarning] = {
+      for {
+        mocksProvider <- mocksProvider(impersonation)
+        warning <- mocksProvider.ldapServiceWith(serviceId) match {
+          case Some(_) => None
+          case None =>
+            ImpersonationWarning(
+              block = blockName,
+              ruleName = ruleName,
+              message = impersonationNotSupportedWithoutMockMsg(ruleName),
+              hint = s"Configure a mock of an LDAP service with ID [${serviceId.value.value}]"
+            ).some
+        }
+      } yield warning
+    }
+
+    private def mocksProvider(impersonation: Impersonation) = impersonation match {
+      case Impersonation.Enabled(settings) => Some(settings.mocksProvider)
+      case Impersonation.Disabled => None
+    }
+
+    private def impersonationNotSupportedWithoutMockMsg(ruleName: Rule.Name): NonEmptyString = NonEmptyString.unsafeFrom(
+      s"The rule '${ruleName.value}' will not match during impersonation until a mock of service is not configured"
```

Comment:
* `will fail to match the impersonating request when the mock ...`
* `of the  '$serviceName' service`